src/ResourceManager/CognitiveServices/CognitiveServices.Test/TestController.cs

```diff
@@ -0,0 +1,190 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Gallery;
+using Microsoft.Azure.Management.Authorization;
+using Microsoft.Azure.Management.Resources;
+using Microsoft.Azure.Management.CognitiveServices;
+using Microsoft.Azure.Subscriptions;
+using Microsoft.Azure.Test;
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.WindowsAzure.Commands.ScenarioTest;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using RestTestFramework = Microsoft.Rest.ClientRuntime.Azure.TestFramework;
+using Microsoft.Azure.ServiceManagemenet.Common.Models;
+
+namespace Microsoft.Azure.Commands.Management.CognitiveServices.Test.ScenarioTests
+{
+    public class TestController
+    {
+        private CSMTestEnvironmentFactory csmTestFactory;
+        private EnvironmentSetupHelper helper;
+
+        public ResourceManagementClient ResourceManagementClient { get; private set; }
+
+        public SubscriptionClient SubscriptionClient { get; private set; }
+
+        public AuthorizationManagementClient AuthorizationManagementClient { get; private set; }
+
+        public CognitiveServicesManagementClient CognitiveServicesClient { get; private set; }
+
+        public GalleryClient GalleryClient { get; private set; }
+
+
+        public string UserDomain { get; private set; }
+
+        public static TestController NewInstance
+        {
+            get
+            {
+                return new TestController();
+            }
+        }
+
+        public TestController()
+        {
+            helper = new EnvironmentSetupHelper();
+        }
+
+        public void RunPsTest(XunitTracingInterceptor traceInterceptor, params string[] scripts)
+        {
+            helper.TracingInterceptor = traceInterceptor;
+            RunPsTest(scripts);
```

Comment:
@daniel-ms this is not safe to do because a call to TestUtilities.GetCallingClass on line  72 and 73. You will need to refactor this in way that it will not depend on call stack depth. This is a known issue in all existing tests and we will need to make a general fix from our side.